src/controllers/ilp-prepare.ts

```diff
@@ -47,19 +47,35 @@ export default class IlpPrepareController {
     log.trace('sending outbound ilp prepare. destination=%s amount=%s', destination, nextHopPacket.amount)
     const result = await outbound(IlpPacket.serializeIlpPrepare(nextHopPacket), nextHop)
 
-    if (result[0] === IlpPacket.Type.TYPE_ILP_FULFILL) {
-      log.trace('got fulfillment. cond=%s nextHop=%s amount=%s', executionCondition.slice(0, 6).toString('base64'), nextHop, nextHopPacket.amount)
-
-      this.backend.submitPayment({
+    if (this.backend.submitPacket) {
+      this.backend.submitPacket({
         sourceAccount: sourceAccount,
         sourceAmount: amount,
         destinationAccount: nextHop,
-        destinationAmount: nextHopPacket.amount
+        destinationAmount: nextHopPacket.amount,
+        parsedPacket,
+        result
+      }).catch(err => {
+        const errInfo = (err && typeof err === 'object' && err.stack) ? err.stack : String(err)
+        log.error('error while submitting packet to backend. error=%s', errInfo)
       })
-        .catch(err => {
-          const errInfo = (err && typeof err === 'object' && err.stack) ? err.stack : String(err)
-          log.error('error while submitting payment to backend. error=%s', errInfo)
+    }
+
+    if (result[0] === IlpPacket.Type.TYPE_ILP_FULFILL) {
+      log.trace('got fulfillment. cond=%s nextHop=%s amount=%s', executionCondition.slice(0, 6).toString('base64'), nextHop, nextHopPacket.amount)
+
+      if (!this.backend.submitPacket) {
```

Comment:
Wouldn't `if (this.backend.submitPayment)` be clearer as a condition?

Reply:
It would be if `submitPayment` was optional, but it is required and that's not something I wanted to change.